src/services/info/feature-info-layer.service.ts

```diff
@@ -0,0 +1,146 @@
+    })
+    this.featureLayer.set('cyLayerType', FEATURE_LAYER_TYPE)
+    this.featureLayer.setStyle(
+      /**
+       * @param {ol.Feature|ol.render.Feature} feature Feature.
+       * @param {number} resolution Resolution.
+       * @return {Array.<ol.style.Style>} Array of styles.
+       */
+      function (feature: FeatureLike) {
+        const lineColor = feature.get('color') || '#ffcc33'
+        const lineWidth = /** @type {number} */ feature.get('width') || 3
+        const defaultStyle = [
+          new Style({
+            fill: new Fill({
+              color: [255, 255, 0, 0.6],
+            }),
+          }),
+          new Style({
+            stroke: new Stroke({
+              color: '#ffffff',
+              width: 5,
+            }),
+          }),
+          new Style({
+            stroke: new Stroke({
+              color: lineColor,
+              width: lineWidth,
+            }),
+          }),
+        ]
+
+        const geometryType = feature.getGeometry()?.getType()
+        return geometryType == 'Point' || geometryType == 'MultiPoint'
+          ? [new Style({ image: image })]
+          : defaultStyle
+      }
+    )
+    this.map.addLayer(this.featureLayer)
+  }
+
+  highlightFeatures(features: FeatureJSON[], fit: boolean): void {
+    if (features !== undefined && features !== null) {
+      if (this.map.getLayers().getArray().indexOf(this.featureLayer) === -1) {
+        this.map.addLayer(this.featureLayer)
+      }
+
+      const encOpt: ReadOptions = {
+        dataProjection: PROJECTION_LUX,
+        featureProjection: this.map.getView().getProjection(),
+      }
+
+      const olFeatures = new GeoJSON().readFeatures(
+        {
+          type: 'FeatureCollection',
+          features: features,
+        },
+        encOpt
+      )
+
+      if (olFeatures.length > 0) {
+        let extent: Extent = olFeatures[0].getGeometry()?.getExtent() || [
+          0, 0, 0, 0,
+        ]
+        for (const curFeature of olFeatures) {
+          const curExtent = curFeature.getGeometry()?.getExtent() || [
+            0, 0, 0, 0,
+          ]
+          extent = extend(extent, curExtent)
+          if (curFeature.getId() == null) {
+            curFeature.setId(undefined)
+          }
+          if (curFeature.getGeometry()?.getType() === 'GeometryCollection') {
+            const geomCollection =
+              curFeature.getGeometry() as GeometryCollection
+            geomCollection
+              .getGeometriesArray()
+              .forEach((geometry: Geometry) => {
+                const newFeature = curFeature.clone()
+                newFeature.setGeometry(geometry)
+                this.featureLayer.getSource()?.addFeature(newFeature)
+              })
+          } else {
+            this.featureLayer.getSource()?.addFeature(curFeature)
+          }
+        }
+        if (fit && extent) {
+          const fitOptions: FitOptions = {
+            size: this.map.getSize() as Size,
+            maxZoom: 17,
```

Comment:
There is a maxzoom in map store that can be set by the config theme, I don't know if this maxZoom in store should be taken into account. (Can be done later)

Reply:
Good catch, I've added it, passing it to the highlight function from the composable to keep the service free of vue stuff.